File: backend/src/003-archivos/tipos/yml/yml.rb
#!/usr/bin/ruby

require "yaml"

# Insertar datos en un archivo .yml

data = {
    "primero" => 1,
    "segundo" => {"hola" => "adios"}
}

File.open("data.yml", "w") do |f|
    f.write(YAML.dump(data))
end


# ====================================================

# Leer datos de un archivo .yml

File.open("fichero.yml", "r") do |f|
    puts YAML.load(f.read)
end

# Fuente:
# https://ruby-doc.org/stdlib-2.1.0/libdoc/yaml/rdoc/YAML.html

File: backend/src/003-archivos/tipos/yml/yml.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import yaml   # pip3 install pyyaml

# Insertar datos en un archivo .yml

data = {
    "primero": 1,
    "segundo": dict(hola="adios")
}

with open("data.yml", "w") as outfile:
    yaml.dump(data, outfile, default_flow_style=False)

# ====================================================

# Leer datos de un archivo .yml

with open("fichero.yml", "r") as infile:
    print(yaml.load(infile))
